src/UbidotsESP8266.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright (c) 2017 Ubidots.
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPY RIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+Made by: ----- María Carlina Hernández ---- Developer at Ubidots Inc 
+         https://github.com/mariacarlinahernandez  
+         ----- Jose Garcia ---- Developer at Ubidots Inc 
+         https://github.com/jotathebest
+*/
+
+#include "UbidotsESP8266.h"
+
+/***************************************************************************
+CONSTRUCTOR
+***************************************************************************/
+
+/**
+ * Constructor.
+ */
+
+Ubidots::Ubidots(char* token, const char* server) {
+    _token = token;
+    _server = server;
+}
+
+/***************************************************************************
+FUNCTIONS TO MANAGE DATA
+***************************************************************************/
+
+/**
+ * This function is to read the Ubidots' command via serial
+ */
+void Ubidots::readData() {
+
+    int i = 0;
+    while (Serial.available() > 0) {
+      this->_command[i++] = (char) Serial.read();
+    }
+
+    i = 0;
+    readServer();
+    memset(this->_command, 0, sizeof(this->_command));
+}
+
+/**
+ * This function establish the connection and send the request to the server
+ */
+uint8_t Ubidots::sendData() {   
+    
+    uint8_t max_retries = 0;
+    uint8_t orderCode = checkCommand();
+
+    if (orderCode > 0) {
+        while (!_client.connected()){
+            _client.connect(SERVER, PORT);
+            max_retries++;
+            if (max_retries>5) {
```

Comment:
add a comment here, it should be something like: "attempts five times to connect"